benches/main.rs

```diff
@@ -64,6 +67,23 @@ pub fn criterion_benchmark(c: &mut Criterion) {
     });
 }
 
+#[cfg(all(feature = "bzimage", any(target_arch = "x86", target_arch = "x86_64")))]
+pub fn criterion_benchmark_bzimage(c: &mut Criterion) {
+    let guest_mem = create_guest_memory();
+    let image = create_bzimage();
```

Comment:
Can we add this `let` and the following `c.bench_function` call to the `x86` specific `criterion_benchmark` (if the previous suggestion about that makes sense), within a block guarded by a  `#[cfg(feature = "bzimage")]`, so we don't have to define the `criterion_benchmark_bzimage` function altogether (and the `criterion_group` variations)?

Reply:
Nope. The `bzimage` benchmark needs to be in another group because it's run for fewer iterations. The number of iterations is defined per group. I expect such cases, where different benchmarks need various run configurations, to show up more often as we add more and more. I couldn't think of any other way of combining this with feature-gating, other than the `nop` bench. Any ideas?

Reply:
Oh I see. In that case, can we rename `criterion_benchmark_nop` to `criterion_benchmark_bzimage`, and put a `cfg` gated call to `x86_64::benchmark_bzimage` inside? It looks like that way we don't need the additional `#[cfg(not(feature = "bzimage"))] criterion_group! { ...` block, since the function effectively turns into nop when the feature is not enabled. Don't have any other ideas :(

Reply:
I like the separate `nop` better because it can stay unchanged for as many features as we gate-bench. It runs nothing for the minimum number of iterations (10). If we get rid of it, we'll run the same nothing but for more iterations when the feature is off (in this case, 100). But, given that we're only doing this for *one* feature, I don't feel strongly for it. How do you see feature-gating benchmarks with more than 1 feature?

Reply:
I think the number of iterations for nothing doesn't matter that much (it doesn't even show up in the results), but yeah it was just an alternative and the current one is perfectly fine as well.